.github/workflows/libcxx-build-and-test.yaml

```diff
@@ -0,0 +1,137 @@
+name: Build and Test libc++
+
+on: pull_request
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number }}
+  cancel-in-progress: true
+
+permissions:
+  checks: write
```

Comment:
Can we set the permissions to read only at the top level of the file and add additional permissions at the job level? That's what we're trying to do across the rest of the project to help reduce the possibility of mistakenly adding permissions where they aren't needed.

Reply:
I think it's vestigial. Removing.

Reply:
Uploading artifacts might need additional permissions. Not completely sure how that is supposed to work.

Reply:
I don't think you need any permissions for uploading artifacts.  We have at least one workflow that does it with read-only permissions:  https://github.com/llvm/llvm-project/blob/main/.github/workflows/libclang-abi-tests.yml

Reply:
All of the permissions have been removed and it seems to work great.